src/server/index.js

```diff
@@ -9,11 +9,124 @@
 const Koa = require('koa');
 const IO = require('koa-socket');
 const Redux = require('redux');
+import Crypto from 'crypto';
+import Router from 'koa-router';
+import koaBody from 'koa-body';
 import { InMemory, Mongo } from './db';
 import { createGameReducer } from '../core/reducer';
+
 const PING_TIMEOUT = 20 * 1e3;
 const PING_INTERVAL = 10 * 1e3;
 
+const isActionFromAuthenticPlayer = async ({
+  action,
+  db,
+  gameID,
+  playerID,
+}) => {
+  const gameMetadata = await db.get(`${gameID}:metadata`);
+  if (!gameMetadata) {
+    return true;
+  }
+
+  const hasSecrets = Object.keys(gameMetadata.players).some(key => {
+    return !!gameMetadata.players[key].secret;
+  });
+  if (!hasSecrets) {
+    return true;
+  }
+
+  if (!action.payload.secret) {
+    return false;
+  }
+
+  if (action.payload.secret !== gameMetadata.players[playerID].secret) {
+    return false;
+  }
+
+  return true;
+};
+
+const createSecret = async () => {
+  return new Promise((resolve, reject) => {
+    Crypto.randomBytes(32, (err, buf) => {
+      if (err) {
+        reject(err);
+      } else {
+        resolve(buf.toString());
+      }
+    });
+  });
+};
+
+const getNewGameInstanceID = ({ gameName }) => {
+  // TODO: Return a new instance ID for this game
+  return gameName !== null;
+};
+
+// HACK: this whole thing needs to move out
+const createApiServer = ({ db, games }) => {
+  const app = new Koa();
+  const router = new Router();
+
+  router.post('/games/:id/create', koaBody(), async ctx => {
+    const gameName = ctx.params.id;
+    const numPlayers = ctx.request.body.numPlayers;
+
+    const newMetaDataState = {
+      players: {},
+    };
+
+    const reducer = createGameReducer({
+      game: games[gameName],
+      numPlayers,
+    });
+    const store = Redux.createStore(reducer);
```

Comment:
This code is copied from the adhoc `sync` game creation. I  left the sync game creation code there because it needs to be there to support non-authed games, but we could probably refactor up the duplication here.